src/sync.d

```diff
@@ -1109,15 +1113,11 @@ final class SyncEngine
 		try {
 			fileDetails = onedrive.getFileDetails(item.driveId, item.id);
 		} catch (OneDriveException e) {
+			log.error("ERROR: Query of OneDrive for file details failed");
 			if (e.httpStatusCode >= 500) {
 				// OneDrive returned a 'HTTP 5xx Server Side Error' - gracefully handling error - error message already logged
 				downloadFailed = true;
 				return;
-			} else {
-				// Default operation if not a 500 error
-				log.error("ERROR: Query of OneDrive for file details failed");
-				downloadFailed = true;
-				return;
```

Comment:
The only thing that puzzles me here is the removal of the else case here. What happens if there is an exception and it is `<500`? Is this impossible?

The rest looks fine to me.

Reply:
So that was added here: https://github.com/abraunegg/onedrive/pull/551/commits/ffc8cd23298becf9e106a5aa516f0f00409983fa

My thoughts are:
* >= 500 will get a downloadFailed handle and return
* < 500 will then generate a JSON object error, which would get handled by line 1125 - `if (fileDetails.type() == JSONType.object){`